### PR TITLE
Invoice class

### PR DESCRIPTION
**Pull request by mzelesky**
This is to create a model of an invoice at the header level. Future work will involve creating a separate class for an invoice line, and a class that retrieves the list of invoices returned from an API call.

### CODE REVIEW

**Comment by mzelesky** [comment]
Lines not currently tested:
Returning an invoice reference number when there is one
Returning invoice notes when there are any

**Comment by mzelesky** [comment]
@sandbergja Can you please review? This is similar to the invoice class used in lib_jobs, but I need a lot more data represented since this is going to be used to perform detailed analysis on invoices and orders.

File: lib/lsp-data/api_invoice.rb [hunk 1]
```diff
@@ -0,0 +1,167 @@
+# frozen_string_literal: true
+
+module LspData
+  ### This class makes a model of an invoice object returned from Alma via API.
+  ###   While it is meant to be a comprehensive model, there are 2 areas that
+  ###     are not of interest to PUL: VAT reporting and explicit currency
+  ###     conversion ratios. Those areas will not be represented.
```
Comment (sandbergja) [inline lib/lsp-data/api_invoice.rb:7]: 👍🏻

**Review by sandbergja — APPROVED** [review]
Looks great, thanks @mzelesky ! 🪈

**Pull request merged**